src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -359,4 +357,59 @@ private ScopeTable newChild() {
       return new ScopeTable(this);
     }
   }
+
+  private static class Scanner {
+    private static final int EOF = -1;
+    private static final int IDENT = -2;
+
+    private final String s;
+    private String token = "";
+    private int p;
+
+    private Scanner(String s) {
+      this.s = s;
+    }
+
+    private int scan() {
+      token = "";
+      int cp;
+
+      while (true) {
+        if (p == s.length()) {
+          return EOF;
+        }
+        cp = s.codePointAt(p);
+        if (!Character.isWhitespace(cp)) {
+          break;
+        }
+        p += Character.charCount(cp);
+      }
+
+      if (identLead(cp)) {
+        int start = p;
+        while (true) {
+          if (p == s.length()) {
+            token = s.substring(start);
+            return IDENT;
+          }
+          cp = s.codePointAt(p);
+          if (!identFollow(cp)) {
+            token = s.substring(start, p);
+            return IDENT;
+          }
+          p += Character.charCount(cp);
+        }
+      }
+      p += Character.charCount(cp);
+      return cp;
+    }
+
+    private static boolean identLead(int codePoint) {
+      return Character.isLetter(codePoint) || "$_".indexOf(codePoint) >= 0;
+    }
+
+    private static boolean identFollow(int codePoint) {
+      return identLead(codePoint) || Character.isDigit(codePoint);
```

Comment:
"$" handling: We want `$` to be an identifier modifier prefix for our own purposes (eg `$resp`). Specifically, my understanding is that we were going to allow it _just_ to be a prefix. If that's not what you had in mind, let's discuss. 
If it is the case, then `$` should not be allowed in this function. As you have it, this allows identifiers such as `$$`, `$foo$bar`, and `$foo$`.

Reply:
Done. Note that `$foo$bar` should then be legal. The string consists of two identifiers, `$foo` and `$bar`.